src/pybind/mgr/dashboard/settings.py

```diff
@@ -236,6 +237,10 @@ def handle_option_command(cmd, inbuf):
     elif cmd['prefix'].startswith('dashboard get'):
         return 0, str(getattr(Settings, opt['name'])), ''
     elif cmd['prefix'].startswith('dashboard set'):
+        if cmd['prefix'] == 'dashboard set-grafana-api-url':
+            if not url_validator(cmd['value']):
+                return (-errno.EINVAL, '', 'invalid value. \'{0}\' is not a valid URL'
+                        .format(cmd['value']))
```

Comment:
@epuertat @votdev @rkachach it seems reasonable to me, if you agree I proceed with implementation.

Reply:
Cool!

Reply:
@epuertat I made a demo commit, in a separated branch, where `Settings` class is removed, as you suggested: https://github.com/torchiaf/ceph/commit/15b1deda1f34471e95df462b5174d7add455c34e

The change is still partial, in the sense that I tested it for few dashboard settings, like `GRAFANA_API_URL`, and although it works fine, there are still `Settings.PROPERTY` statements all over the code, which should be converted to `mgr.get_module_option('PROPERTY')` before to cherry-pick the commit to this PR.

Before to do it, I wanted to ensure the change is still convenient.
In particular:

- `dashboard/settings.py` and `dashboard/module.py` files are heavily refactored, I don't know if this could be too risky, considering the settings in there are used all over the dashboard module.
- there are a ton of files which need to chage  `Settings.PROPERTY` to `mgr.get_module_option('PROPERTY')`. I don't know if this could easily introduce some regressions.

What are you thoughts? Please take a look when you can, thanks.

Reply:
@torchiaf @epuertat Can we do that refactoring in a separate PR? The feature that is implemented here has nothing to do with this. A separate PR will also make it easier to backport this feature. I don't think we want to have that refactoring in Pacific for example.

Reply:
@epuertat @votdev @torchiaf Sorry guys but I missed this conversation. I think having support for parameters validation on the mgr could be beneficial for everybody and we can have generic validators for common data types and custom ones if we need to. I have it on my TODO list but unfortunately was very busy so far with other stuff so I didn't get the time to work on it.